WeakAuras/GenericTrigger.lua

```diff
@@ -3392,8 +3393,57 @@ function WeakAuras.WatchUnitChange(unit)
     watchUnitChange:RegisterEvent("UNIT_PET")
     watchUnitChange:RegisterEvent("RAID_TARGET_UPDATE")
 
+    local function testMultiUnitIsUnit(unitA, unitB)
```

Comment:
This does a lot of checks, quite a lot more than I would want to add. Fortunately I think there's an easier way to handle that if we have two datastructures:

unitIdToGUID map (which we alreaedy have in watchUnitChange.unitChangeGUIDS
and GUIDToUnitIds map

And we rely on UnitGUID returning the same value if two unit ids refer to the same unit.

Then e.g. the PLAYER_TARGET_CHANGED code becomes:
1)
local eventsToSend = {}

local oldguid = unitIdToGUID["target"]
if oldguid then
  for unitId in pairs(GUIDToUnitIds[oldguid]) do
    if unitId ~= "target" then
      -- Check for multi units here? Unsure
     eventsToSend["UNIT_IS_UNIT_CHANGED_target_" .. unitId] = target
     eventsToSend["UNIT_IS_UNIT_CHANGED_" .. unitID .. "_target"] = unitId
   end
  end
end

local newguid = UnitGUID("target")
if newguid then
  if GUIDToUnitIds["target"] then
    for unitId in pairs(GUIDToUnitIds[newguid]) do
      -- Check for multi units here? Unsure
      eventsToSend["UNIT_IS_UNIT_CHANGED_target_" .. unitId] = target
      eventsToSend["UNIT_IS_UNIT_CHANGED_" .. unitId .. "_target"] = unitId
    end
  end
end

2) Update data structures
Update watchUnitChange.unitChangeGUIDS
GUIDToUnitIds[oldGUID]["target"] = nil
GUIDToUnitIds[newGUID]["target"] = true

3) Send events
for event, unit in pairs(eventsToSend) do
  WeakAuras.ScanEvents(event, unit)
end


Btw, the code in watchUnitChange OnEvent handler looks a bit strange.
The events:
PLAYER_SOFT_ENEMY_CHANGED, PLAYER_SOFT_FRIEND_CHANGED, INSTANCE_ENCOUNTER_ENGAGE_UNIT, GROUP_ROSTER_UPDATE all end up in the last block, which looks wrong to me.

Reply:
I refactored the whole function

- UnitIsUnit between nameplates & target still work
- Markers on nameplates work
- i don't remember why UNIT_TARGET is register and what to do with it

More tests needs to be done to be sure everything is ok